packages/terra-date-picker/src/react-datepicker/index.jsx

```diff
@@ -593,34 +601,42 @@ class DatePicker extends React.Component {
       let newSelection
       switch (eventKey) {
         case 'ArrowLeft':
+          this.setState({ calendarOpenedViaKeyboard: true })
```

Comment:
We should rename `calendarOpenedViaKeyboard` to something that better represents what its purpose it. It can cause confusion to set this to true in `handleCalendarKeyDown` to indicate the calendar was opened using a keyboard when it may have been opened by clicking on the calendar button.

The way I understand how the focus should work (after talking to @neilpfeiffer) is that the preselected day should show the focus style whenever the user interacts with the picker using the keyboard. Whenever the user interacts by clicking, then the focus on the preselected day should go away. So it is not just at the time when the picker is launched. Therefore, we should name this something along the line of `isCalendarKeyboardFocused` similar to what's in day.jsx.

Reply:
Changed e4930e6